index.html

```diff
@@ -89,46 +121,142 @@
 <script>
   var data = [];
   var options = {
-    yearRange: 'all'
+    Contract_DateRange: 'all'
   };
 
-  var svg;
+  var svg, x, y;
+
+  var margin = {top:20, right:10, bottom:20, left:10}
+    padding = {top: 60, right: 60, bottom: 60, left:60};
+
+
+  var width= 680 - margin.left - margin.right;
+  var height= 400 - margin.top - margin.bottom;
 
   // FETCH DATA
 
+  d3.json('data.json',function(error,json){
+    if (error){throw error;}
+
+    data=json;
+
+    setup();
+    update();
+  });
+  //call back function
+console.log()
+ 
   function setup() {
     // SVG
 
-    svg = d3.select('#chart')
-      .append('svg');
+  var svg = d3.select('#chart')
```

Comment:
This is the problem. Hopefully it's becoming clearer after this week's class demo, but `svg` is a global variable. We define it outside of `setup()` on line 127 so that it's available anywhere in our script. But when you use the `var` keyword here, you create a new _local_ variable with the same name, and this variable expires once `setup()` is finished executing. This is why why you try to use it inside `update()` you get the error about it being undefined: the global `svg` variable was never given a value, because this local `svg` swallowed it.